### PR TITLE
Fix: resolves #1579, #1531

### LINKED ISSUE #146: Oberon takes 5 people to enter
oberon

### LINKED ISSUE #1579: Oberon takes 5 people to enter
oberon

### PR DESCRIPTION
**Pull request by marcosvf132**
This PR contains:
- FIx on falcon teleport to oberon. Add check on teleport if player can face oberon again.

- Fix oberon lever, now player can try to solo oberon.

Resolves #1579 #1531

### CODE REVIEW

**Comment by marcosvf132** [comment]
fix #1579

File: data/movements/scripts/teleport/falcon_castle.lua [hunk 1]
```diff
@@ -1,20 +1,28 @@
...
-	if teleport then
-		player:teleportTo(teleport)
-		fromPosition:sendMagicEffect(CONST_ME_TELEPORT)
-		teleport:sendMagicEffect(CONST_ME_TELEPORT)
+	for index, value in pairs(config) do
+		if item.actionid == index then
+			if(item.actionid == 57604)then
+				if(player:getStorageValue(Storage.TheSecretLibrary.TheOrderOfTheFalcon.OberonTimer) > os.time())then
+					player:sendTextMessage(MESSAGE_EVENT_ADVANCE, "You have faced this boss in the last 20 hours.")
+					player:teleportTo(Position(fromPosition), false)
```
Comment (dudantas) [inline data/movements/scripts/teleport/falcon_castle.lua]: You can only use "fromPosition, true"

File: data/movements/scripts/teleport/falcon_castle.lua [hunk 2]
```diff
@@ -1,20 +1,28 @@
...
+					doSendMagicEffect(Position(fromPosition), CONST_ME_TELEPORT)
```
Comment (dudantas) [inline data/movements/scripts/teleport/falcon_castle.lua]: You can use "player:getPosition():sendMagicEffect(CONST_ME_TELEPORT)

File: data/actions/scripts/farmine/oberon_lever.lua [hunk 3]
```diff
@@ -1,60 +1,51 @@
...
+	local players = {}
+	for i = 0, 4 do
+		local player1 = Tile({x = (Position(item:getPosition()).x - 2) + i, y = Position(item:getPosition()).y + 1, z = Position(item:getPosition()).z}):getTopCreature()
+		players[#players+1] = player1
+	end
+		for i, player in ipairs(players) do
+			player:getPosition():sendMagicEffect(CONST_ME_POFF)
+			player:teleportTo(Position(setting.playerTeleport), false)
+			doSendMagicEffect(player:getPosition(), CONST_ME_TELEPORT)
+			setPlayerStorageValue(player,setting.storage, os.time() + 72000)
```
Comment (thatmichaelguy) [inline data/actions/scripts/farmine/oberon_lever.lua]: The other lever scripts are using form `os.time() + 20 * 60 * 60` which I find easier to understand at a glance.

File: data/actions/scripts/farmine/oberon_lever.lua [hunk 4]
```diff
@@ -1,60 +1,51 @@
...
+				addEvent(function()
+					local spectatorsOberon = Game.getSpectators(Position(setting.centerRoom), false, false, 10, 10, 10, 10)
+						for i = 1, #spectatorsOberon, 1 do
+							if spectatorsOberon[i]:isPlayer() and (spectatorsOberon[i]:getName() == player:getName()) then
+								player:teleportTo(Position(setting.kickPosition))
+								player:getPosition():sendMagicEffect(CONST_ME_TELEPORT)
+								player:sendTextMessage(MESSAGE_EVENT_ADVANCE, 'Time is over.')
+							end
+						end
+				end, 1200000)
```
Comment (thatmichaelguy) [inline data/actions/scripts/farmine/oberon_lever.lua]: If we can change this to `20 * 60 * 1000` it would increase readability

**Comment by thatmichaelguy** [comment]
I'm running this PR in production without issues, lgtm except for the minor readability comments.

**Comment by marcosvf132** [comment]
I know that understand wich one is in hour or minutes is a bit hard in a full number, but the boss time are pretty standard for all boss, so i don't see any problem with that, but i understand you point of view, sometimes a more intuitive script can be better for the user.

File: data/actions/scripts/farmine/oberon_lever.lua (hunk 3 above)
Comment (majestyotbr) [reply data/actions/scripts/farmine/oberon_lever.lua]: I agree.

File: data/actions/scripts/farmine/oberon_lever.lua (hunk 4 above)
Comment (majestyotbr) [reply data/actions/scripts/farmine/oberon_lever.lua]: I agree.

**Comment by thatmichaelguy** [comment]
The boss is not removed with the room clean, if one player/team fails to kill him the lever will return `Someone is fighting against the boss! You need wait awhile.` until he is manually removed.

**Comment by marcosvf132** [comment]
I see where is the problem, the problem is not with room clean, the problem is with the server function roomIsOccupied(). I don't understand why this function is set to return false if there is any creature inside the room, a more useful function would be making this one with more option to let user choose between `onlyplayer` or not.

File: data/actions/scripts/farmine/oberon_lever.lua [hunk 5]
```diff
@@ -1,60 +1,53 @@
...
+	end
+		for i, player in ipairs(players) do
+			player:getPosition():sendMagicEffect(CONST_ME_POFF)
+			player:teleportTo(Position(setting.playerTeleport), false)
+			doSendMagicEffect(player:getPosition(), CONST_ME_TELEPORT)
+			setPlayerStorageValue(player,setting.storage, os.time() + 20 * 60 * 60)
+            player:sendTextMessage(MESSAGE_EVENT_ADVANCE, 'You have 20 minute(s) to defeat the boss.')
+				addEvent(function()
+					local spectatorsOberon = Game.getSpectators(Position(setting.centerRoom), false, false, 10, 10, 10, 10)
+						for i = 1, #spectatorsOberon, 1 do
```
Comment (majestyotbr) [inline data/actions/scripts/farmine/oberon_lever.lua]: Check warning on line 42 in data/actions/scripts/farmine/oberon_lever.lua


GitHub Actions
/ reviewers

data/actions/scripts/farmine/oberon_lever.lua#L42
[luacheck] reported by reviewdog 🐶
shadowing upvalue loop variable 'i' on line 34

Raw Output:
data/actions/scripts/farmine/oberon_lever.lua:42:11: shadowing upvalue loop variable 'i' on line 34

**Review by majestyotbr — APPROVED** [review]

**Pull request merged**